Makefile

```diff
@@ -20,3 +37,70 @@ test:
 install:
 	go install -v ./pkg/... ./cmd/...
 
+.PHONY: clean
+clean:
+	rm -rf $(BUILD_DIR)
+
+.PHONY: build
+build: clean build-darwin-amd64 build-linux-amd64 build-windows-amd64
+
+.PHONY: build-darwin-amd64
+build-darwin-amd64: build-apiregister-gen-darwin-amd64 build-apiserver-boot-darwin-amd64 build-apiserver-builder-release-darwin-amd64
+
+.PHONY: build-apiregister-gen-darwin-amd64 build-apiserver-boot-darwin-amd64 build-apiserver-builder-release-darwin-amd64
+build-apiregister-gen-darwin-amd64 build-apiserver-boot-darwin-amd64 build-apiserver-builder-release-darwin-amd64: build-%-darwin-amd64:
+	mkdir -p $(DARWIN_AMD64_BUILD_BIN_DIR)
+	GOOS=darwin GOARCH=amd64 go build -o $(DARWIN_AMD64_BUILD_BIN_DIR)/$* ./cmd/$*/main.go
+
+.PHONY: build-linux-amd64
+build-linux-amd64: build-apiregister-gen-linux-amd64 build-apiserver-boot-linux-amd64 build-apiserver-builder-release-linux-amd64
+
+.PHONY: build-apiregister-gen-linux-amd64 build-apiserver-boot-linux-amd64 build-apiserver-builder-release-linux-amd64
+build-apiregister-gen-linux-amd64 build-apiserver-boot-linux-amd64 build-apiserver-builder-release-linux-amd64: build-%-linux-amd64:
+	mkdir -p $(LINUX_AMD64_BUILD_BIN_DIR)
+	GOOS=linux GOARCH=amd64 go build -o $(LINUX_AMD64_BUILD_BIN_DIR)/$* ./cmd/$*/main.go
```

Comment:
We may need to do something more than this.  The tar release contains builds of the vendored code-generators + the vendored go packages.

Reply:
Take a look at the artifacts in the .tar.gz published in the release and the release cmd.

Reply:
Understood (see the comment above). The idea is then to run `apiserver-builder-release` to build a `.tar.gz` and use this to create the packages. Do you have any idea why `apiserver-boot` fails when I run `apiserver-builder-release` (again, see the comment above).

Reply:
Hm.  That is weird, the release command should [set the GOPATH](https://github.com/kubernetes-incubator/apiserver-builder/blob/master/cmd/apiserver-builder-release/main.go#L173).  Looking that the code, it looks like it might be just appending the GOPATH.  Perhaps if your GOPATH is already set, it won't override it.  What OS are you running this on?  I have only run it from a Mac.

I am going to try a fix by removing the existing GOPATH env from the `exec`'ed command environment if it is preset.

Reply:
I am running this on a Mac with a GOPATH set. Since you're already on it, I won't open an issue for it then. Thanks!

Reply:
Hm, that is odd.  What shell are you using?  I made some changes so it will prune the GOPATH from the environment when execing.  Try running again now.

Reply:
Your may revert your change to GOPATH handling. The error was due to `glide` being missing.